model/lifecycle.go

```diff
@@ -57,6 +57,7 @@ func cacheFromTask(t task.Task) build.TaskCache {
 		StartTime:     t.StartTime,
 		TimeTaken:     t.TimeTaken,
 		Activated:     t.Activated,
+		Blocked:       t.Blocked(),
```

Comment:
I think a task will only be `Blocked()` when one of its dependencies (or its dependencies dependencies, etc...) [has run and failed](https://github.com/evergreen-ci/evergreen/blob/4634cc2fa643e23da573e68d1fd8e281da0a0fa3/model/task_lifecycle.go#L440). 
It looks to me like the task is cached before that happens and not necessarily updated after the task has been blocked.

Reply:
We could probably update the task cache for dependent tasks in UpdateBlockedDependencies and that won't be too expensive? @tychoish what do you think? (This is a kind of massive refactor bc of import cycles, stay tuned)